<launch>
  <arg name="project_file" default=""/>
  <arg name="model" default=""/>
  <arg name="control_config" default=""/>
  <arg name="robot_name" default="AssembleRobot"/>
  <arg name="controllers" default="joint_controller" />

  <!-- Choreonoid on ROS -->
  <!-- TODO choreonoid_rosがinstall dirへlaunch fileコピーをコピーするようになった際にfindコマンドを使うようにする．-->
  <!-- include file="$(find choreonoid_ros)/launch/choreonoid.launch"-->
  <include file="/choreonoid_ws/src/choreonoid_ros/launch/choreonoid.launch">
    <arg name="choreonoid_arg" value="$(arg project_file)"/>
  </include>

  <!-- ROS Control -->
  <group ns="$(arg robot_name)">
    <rosparam file="$(arg control_config)" command="load"/>
    <!-- robot description -->
    <param name="robot_description" command="$(find xacro)/xacro '$(arg model)'"/>

    <!-- state publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true"/>

    <!-- load the controllers -->
    <node name="robot_controller_spawner" pkg="controller_manager"
          type="spawner" output="screen" respawn="true"
          args="$(arg controllers)"/>
  </group>
</launch>
